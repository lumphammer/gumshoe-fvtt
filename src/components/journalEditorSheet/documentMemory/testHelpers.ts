import { vi } from "vitest";

export const epoch = 0;

export function getAdditiveStates(count: number): string[] {
  if (count < 1 || count > 26) {
    throw new Error("count must be between 1 and 26");
  }
  const result: string[] = [];
  // loop over the indices of the final array
  for (let i = 0; i < count; i++) {
    const state = new Array(i + 1)
      .fill(null)
      .map((_, i) => String.fromCharCode(i + 97));
    result.push(state.join(""));
  }
  return result;
}

export function advanceTime10s(): void {
  vi.setSystemTime(new Date().setSeconds(new Date().getSeconds() + 10));
}
