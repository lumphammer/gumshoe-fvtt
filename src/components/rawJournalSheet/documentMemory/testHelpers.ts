import { vi } from "vitest";

export const epoch = 0;

const twentyStates = [
  "a",
  "ab",
  "abc",
  "abcd",
  "abcde",
  "abcdef",
  "abcdefg",
  "abcdefgh",
  "abcdefghi",
  "abcdefghij",
  "abcdefghijk",
  "abcdefghijkl",
  "abcdefghijklm",
  "abcdefghijklmn",
  "abcdefghijklmno",
  "abcdefghijklmnop",
  "abcdefghijklmnopq",
  "abcdefghijklmnopqr",
  "abcdefghijklmnopqrs",
  "abcdefghijklmnopqrst",
];

export const getStates = (n: number) => twentyStates.slice(0, n);

export function advanceTime10s() {
  vi.setSystemTime(new Date().setSeconds(new Date().getSeconds() + 10));
}
