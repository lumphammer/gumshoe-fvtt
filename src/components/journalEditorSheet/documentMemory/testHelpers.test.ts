import { describe, expect, test } from "vitest";

import { getAdditiveStates } from "./testHelpers";

describe("getAdditiveStates", () => {
  test.each(Array.from({ length: 9 }, (_, i) => i + 1))(
    "getStates: %i states",
    (i) => {
      expect(getAdditiveStates(i)).toMatchSnapshot();
    },
  );
  test.each([0, -1, -100, 27, 100])(
    "getStates: %i states should throw",
    (i) => {
      expect(() => getAdditiveStates(i)).toThrowError(
        "count must be between 1 and 26",
      );
    },
  );
});

describe("Vitest expect.closeTo", () => {
  test("0.1 + 0.2", () => {
    const result = 0.1 + 0.2;
    expect(result).not.toEqual(0.3);
    expect(result).toEqual(expect.closeTo(0.3, 5));
  });
  test("0.3 + 0.6", () => {
    const result = 0.3 + 0.6;
    expect(result).not.toEqual(0.9);
    expect(result).toEqual(expect.closeTo(0.9, 5));
  });
});
