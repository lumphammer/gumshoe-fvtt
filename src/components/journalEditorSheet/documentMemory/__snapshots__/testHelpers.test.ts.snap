// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getAdditiveStates > getStates: 1 states 1`] = `
[
  "a",
]
`;

exports[`getAdditiveStates > getStates: 2 states 1`] = `
[
  "a",
  "ab",
]
`;

exports[`getAdditiveStates > getStates: 3 states 1`] = `
[
  "a",
  "ab",
  "abc",
]
`;

exports[`getAdditiveStates > getStates: 4 states 1`] = `
[
  "a",
  "ab",
  "abc",
  "abcd",
]
`;

exports[`getAdditiveStates > getStates: 5 states 1`] = `
[
  "a",
  "ab",
  "abc",
  "abcd",
  "abcde",
]
`;

exports[`getAdditiveStates > getStates: 6 states 1`] = `
[
  "a",
  "ab",
  "abc",
  "abcd",
  "abcde",
  "abcdef",
]
`;

exports[`getAdditiveStates > getStates: 7 states 1`] = `
[
  "a",
  "ab",
  "abc",
  "abcd",
  "abcde",
  "abcdef",
  "abcdefg",
]
`;

exports[`getAdditiveStates > getStates: 8 states 1`] = `
[
  "a",
  "ab",
  "abc",
  "abcd",
  "abcde",
  "abcdef",
  "abcdefg",
  "abcdefgh",
]
`;

exports[`getAdditiveStates > getStates: 9 states 1`] = `
[
  "a",
  "ab",
  "abc",
  "abcd",
  "abcde",
  "abcdef",
  "abcdefg",
  "abcdefgh",
  "abcdefghi",
]
`;
